Tools/WinMLRunner/src/BindingUtilities.h

```diff
@@ -127,26 +126,46 @@ template <> struct TensorKindToValue<TensorKind::String>
     typedef TensorString Type;
 };
 
+void GetHeightAndWidthFromLearningModelFeatureDescriptor(const ILearningModelFeatureDescriptor& modelFeatureDescriptor,
+                                                         uint64_t& width, uint64_t& height)
+{
+    if (modelFeatureDescriptor.Kind() == LearningModelFeatureKind::Tensor)
+    {
+        // We assume NCHW
+        auto tensorDescriptor = modelFeatureDescriptor.try_as<TensorFeatureDescriptor>();
+        if (tensorDescriptor.Shape().Size() != 4)
+        {
+            throw hresult_invalid_argument(L"Cannot generate arbitrary image for tensor input of dimensions: " +
+                                           tensorDescriptor.Shape().Size());
+        }
+        height = tensorDescriptor.Shape().GetAt(2);
+        width = tensorDescriptor.Shape().GetAt(3);
+    }
+    else if (modelFeatureDescriptor.Kind() == LearningModelFeatureKind::Image)
+    {
+        auto imageDescriptor = modelFeatureDescriptor.try_as<IImageFeatureDescriptor>();
+        height = imageDescriptor.Height();
+        width = imageDescriptor.Width();
+    }
+    else
+    {
+        throw hresult_not_implemented(
+            L"Generating arbitrary image not supported for input types that aren't tensor or image.");
+    }
+}
+
 namespace BindingUtilities
 {
     static unsigned int seed = 0;
     static std::independent_bits_engine<std::default_random_engine, CHAR_BIT, unsigned int> randomBitsEngine;
 
-    SoftwareBitmap GenerateGarbageImage(const TensorFeatureDescriptor& imageDescriptor, InputDataType inputDataType)
+    SoftwareBitmap GenerateGarbageImage(const ILearningModelFeatureDescriptor& modelFeatureDescriptor,
```

Comment:
It seems the function is assuming the input descriptor is imagefeature. Should we narrow the scope of this input parameter to just accept ImageFeatureDescriptor?

Reply:
Never mind this comment as this can be a TensorFeatureDescriptor